cmd/monaco/main.go

```diff
@@ -61,7 +63,7 @@ func RunImpl(args []string, fileReader util.FileReader) (statusCode int) {
 
 	statusCode = 0
 
-	dryRun, verbose, environments, projectNameToDeploy, path, errorList, flagError := parseInputCommand(args, fileReader)
+	dryRun, verbose, environments, projectNameToDeploy, downloadFlag, downloadSpecificAPI, path, errorList, flagError := parseInputCommand(args, fileReader)
```

Comment:
Might make sense to create a `struct` which holds all the results. Retuning 9 arguments here becomes very hard to read.

Reply:
this was trickier, I have created the struct and it works fine but I couldn't implement a unit test for this one without refactoring a lot of the method, the flag.Parse() should be call outside this method for it to be able to be unit tested https://stackoverflow.com/questions/38711653/unit-test-different-flag-values or https://eli.thegreenplace.net/2020/testing-flag-parsing-in-go-programs/. I think that the code in main for the upload functionality should be moved to a new class and main should be refactor. If nobody is working on it I can create a new pull request in a couple of weeks to do this. Also saw the new experimental CLI but this would break the merge of this funcionality and it's still leaving upload code on main.go file